tests/jballerina-unit-test/src/test/resources/test-src/bala/test_bala/annotations/annotation.bal

```diff
@@ -27,6 +27,28 @@ function testAnnotOnBoundMethod() {
     assertEquality(102, rec.i);
 }
 
+type AnnotationRecord record {|
+    int minValue;
+    int maxValue;
+|};
+
+annotation AnnotationRecord BarAnnotation on type ;
+
+@BarAnnotation {
+    minValue: 18,
+    maxValue: 36
+}
+type Bar int|float;
+
+function testAnnotationWithReferredType() {
+    typedesc<any> 'type = typeof Bar;
+    AnnotationRecord? 'annotation = 'type.@BarAnnotation;
+    if ('annotation is AnnotationRecord) {
+        assertEquality('annotation.minValue, 18);
+        assertEquality('annotation.minValue, 36);
+    }
```

Comment:
This test should fail if `'annotation` is `()`.

Reply:
Updated.